src/west/commands/project.py

```diff
@@ -787,9 +798,11 @@ def _update(project, fetch, rebase, keep_descendants):
 def _clone(project):
     log.small_banner(project.format('{name}: cloning and initializing'))
     project.git('init {abspath}', cwd=util.west_topdir())
-    # This remote is only added for the user's convenience. We always fetch
-    # directly from the URL specified in the manifest.
-    project.git('remote add -- {remote_name} {url}')
+    # The "origin" remote is only added for the user's convenience. We
+    # always fetch directly from the URL specified in the manifest.
+    # Later changes to the manifest won't change this remote,
+    # unfortunately -- the user will have to fix that themselves.
+    project.git('remote add -- origin {url}')
```

Comment:
I know it's a big breaking change, and I know that will take some doc rewriting, but I don't have a better alternative.

We need to get rid of this because I believe there's no clean way to allow manifest imports with first-class remotes. If nrf/west.yml defines a remote R with url-base-A and imports zephyr/west.yml, which defines R to have url-base-B, there's no clean way to do this as an "override".

Projects in zephyr/west.yml are completely "inlined" *before* being imported into nrf/west.yml. At that point, the remotes disappear and the URL is what really matters.

If you look at the model for how manifest imports work here, you can see that only a subset of project data is considered for merging when we model them as tuples:

file:///home/mbolivar/zp/zephyr/doc/_build/html/guides/west/manifest.html#west-manifest-formal

The 'remote' is not part of that.

I did it this way because if we have both `remote` and `url` attributes in a west.yml entry for a project, there's no clear way to decide which one should win if a sub-manifest uses one, and a parent manifest uses the other.

So I basically "erase" the remotes after parsing each manifest and just override the URLs.

Reply:
Not sure I understand this, or even why this had to be introduced:
https://github.com/zephyrproject-rtos/west/pull/329/files#r344823583

cause we only set the remote during `_clone` which only happens on first invocation of `west update` after init or when a new project is added to the manifest file.
https://github.com/mbolivar/west/blob/manifest-imports/src/west/commands/project.py#L691-L692

After that point we never touches the remote again, and I don't see this PR changes that behavior.

Reply:
This was needed due to the original 'override' semantics. It's probably no longer needed in the new 'ignore' semantics you and Carles have suggested, but the code remained. We can look into adding them back in as follow-up work.